pkg/crc/machine/start.go

```diff
@@ -112,41 +112,104 @@ func (client *client) updateVMConfig(startConfig types.StartConfig, vm *virtualM
-		return fmt.Errorf("Unexpected root device: %s", rootPart)
+		return "", fmt.Errorf("Unexpected root device: %s", rootPart)
 	}
+	return rootPart, nil
+}
+
+func runGrowpart(sshRunner *crcssh.Runner, rootPart string) error {
+	var exitErr *ssh.ExitError
 	// with '/dev/[sv]da4' as input, run 'growpart /dev/[sv]da 4'
-	if _, _, err := sshRunner.RunPrivileged(fmt.Sprintf("Growing %s partition", rootPart), "/usr/bin/growpart", rootPart[:len("/dev/.da")], rootPart[len(rootPart)-1:]); err != nil {
-		var exitErr *ssh.ExitError
-		if !errors.As(err, &exitErr) {
-			return err
-		}
-		if exitErr.ExitStatus() != 1 {
-			return err
-		}
+	_, _, err := sshRunner.RunPrivileged(fmt.Sprintf("Growing %s partition", rootPart), "/usr/bin/growpart", rootPart[:len("/dev/.da")], rootPart[len(rootPart)-1:])
+	if !errors.As(err, &exitErr) {
+		return err
+	}
+	if exitErr.ExitStatus() != 1 {
+		return err
+	}
 
-		logging.Debugf("No free space after %s, nothing to do", rootPart)
-		return nil
+	logging.Debugf("No free space after %s, nothing to do", rootPart)
+	return nil
+}
+
+func growLVForMicroshift(sshRunner *crcssh.Runner) error {
+	rootPart, err := getrootPartition(sshRunner, "/dev/disk/by-id/lvm-pv-*")
+	if err != nil {
+		return err
+	}
+	if err := runGrowpart(sshRunner, rootPart); err != nil {
+		return err
 	}
 
-	logging.Infof("Resizing %s filesystem", rootPart)
-	if _, _, err := sshRunner.RunPrivileged("Remounting /sysroot read/write", "mount -o remount,rw /sysroot"); err != nil {
+	if _, _, err := sshRunner.RunPrivileged("Resizing the physical volume(PV)", "/usr/sbin/pvresize", rootPart); err != nil {
 		return err
 	}
-	if _, _, err = sshRunner.RunPrivileged(fmt.Sprintf("Growing %s filesystem", rootPart), "xfs_growfs", rootPart); err != nil {
+
+	// Get the size of volume group
+	// $ sudo vgs --noheadings --nosuffix --units g -o vg_size
+	//   39.02
+	sizeVG, _, err := sshRunner.RunPrivileged("Get the volume group size", "/usr/sbin/vgs", "--noheadings", "--nosuffix", "--units", "g", "-o", "vg_size")
+	if err != nil {
+		return err
+	}
+	vgSize, err := strconv.ParseFloat(strings.TrimSpace(sizeVG), 64)
+	if err != nil {
 		return err
 	}
 
+	// Get the free space for volume group
+	// $ sudo vgs --noheadings --nosuffix --units g -o vg_free
+	//   7.51
+	freeVG, _, err := sshRunner.RunPrivileged("Get the free space for volume group", "/usr/sbin/vgs", "--noheadings", "--nosuffix", "--units", "g", "-o", "vg_free")
+	if err != nil {
+		return err
+	}
+	vgFree, err := strconv.ParseFloat(strings.TrimSpace(freeVG), 64)
+	if err != nil {
+		return err
+	}
+
+	// We want to expand the `/dev/rhel/root` vg in case the free size of volume group is greater than half of total size
+	// since that vg contain the `/sysroot` so we need to extend it to accommodate more container images. Free size is
+	// used by logical volume manager storage (LVMS) Container Storage Interface (CSI) provider for storing PVs.
+	sizeDiff := vgFree - vgSize/2
```

Comment:
Having a hard time to wrap my head around this, might be easier to read as (which should be approximately correct?)
```
  if vgFree > vgSize/2 {
    // run lvextend --resizefs --size $vgSize/2
  }
```